.github/workflows/otelcol-fips.yml

```diff
@@ -27,7 +27,7 @@ env:
 
 jobs:
   otelcol-fips:
-    runs-on: ${{ fromJSON('["ubuntu-24.04", "otel-arm64"]')[matrix.GOARCH == 'arm64'] }}
+    runs-on: ${{ fromJSON('["ubuntu-24.04", "ubuntu-24.04-arm"]')[matrix.GOARCH == 'arm64'] }}
```

Comment:
Why do we move to another image here?

Reply:
@pjanotti [suggested it above](https://github.com/signalfx/splunk-otel-collector/pull/6329#discussion_r2155326373). I'm not very familiar with the nuances of the otel arm runner, but my thought in changing as suggested was that it will be an easier maintenance burden going forward to use the GitHub runner.

Reply:
I'll revert this change and we can revisit at a later date.